pkg/build/graph/analysis/bc.go

```diff
@@ -98,3 +102,85 @@ func FindCircularBuilds(g osgraph.Graph) []osgraph.Marker {
 
 	return markers
 }
+
+// FindPendingTags inspects all imageStreamTags that serve as outputs to builds.
+//
+// Precedence of failures:
+// 1. A build config points to the non existent tag but no current build exists.
+// 2. A build config points to the non existent tag but the latest build has failed.
+func FindPendingTags(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
+		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+		if bcNode, points := buildPointsToTag(g, uncastIstNode); points && !istNode.Found() {
+			buildNode := latestBuild(g, bcNode)
+
+			// A build config points to the non existent tag but no current build exists.
+			if buildNode == nil {
+				markers = append(markers, osgraph.Marker{
+					Node:         bcNode,
+					RelatedNodes: []graph.Node{uncastIstNode},
+
+					Severity:   osgraph.WarningSeverity,
+					Key:        TagNotAvailableWarning,
+					Message:    fmt.Sprintf("%s needs to be imported or created by a build.", istNode.ResourceString()),
+					Suggestion: osgraph.Suggestion(fmt.Sprintf("oc start-build %s", bcNode.(*buildgraph.BuildConfigNode).ResourceString())),
+				})
+				continue
+			}
+
+			// A build config points to the non existent tag but something is going on with
+			// the latest build.
+			// TODO: Handle other build phases.
+			switch buildNode.(*buildgraph.BuildNode).Build.Status.Phase {
+			case buildapi.BuildPhaseCancelled:
+				// TODO: Add a warning here.
+			case buildapi.BuildPhaseError:
+				// TODO: Add a warning here.
+			case buildapi.BuildPhaseComplete:
+				// We should never hit this. The output of our build is missing but the build is complete.
+				// Most probably the user has messed up?
+				// TODO: What should we do here?
```

Comment:
If we can determine that something is wrong, we should expose that to the user, who is probably at this point wondering why their deployment isn't working. If this code is racy and that causes us to hit this case, then we need to fix that race ...